soot-infoflow/src/soot/jimple/infoflow/memory/MemoryWarningSystem.java

```diff
@@ -191,8 +191,17 @@ public void setWarningThreshold(double percentage) {
 			logger.info(MessageFormat.format("Registered a memory warning system for {0} MiB",
 					(threshold / 1024D / 1024D)));
 			warningSystems.add(this);
-			tenuredGenPool.setUsageThreshold(warningSystems.iterator().next().threshold);
-			if (!tenuredGenPool.isUsageThresholdSupported()) {
+			MemoryUsage usage = tenuredGenPool.getUsage();
+			long threshold = warningSystems.iterator().next().threshold;
+			boolean useOwnImplementation = !tenuredGenPool.isUsageThresholdSupported();
+			if (!useOwnImplementation && usage != null && usage.getUsed() > threshold) {
+				tenuredGenPool.setUsageThreshold(threshold);
+			} else {
+				// when the usage is already above the threshold, we use our own implementation,
+				// since the jvm implementation does not seem to get called in that case.
+				useOwnImplementation = true;
```

Comment:
Warum? Hier kommt man nur rein, wenn ``useOwnImplementation`` bereits ``true`` ist

Reply:
Auch wenn sie false ist und der threshold unter dem aktuellen Verbrauch ist, kommt man in den Branch.